Source/LinkedList.c

```diff
@@ -2,5 +2,48 @@
  * @file
  * @brief Implementation of singly-linked list.
  */
-
+#include <stdio.h>
+#include <stdlib.h>
 #include "LinkedList.h"
+
+
+void LinkedList_Init(LinkedList_t *instance)
+{
+   instance->head = NULL;
+}
+
+int LinkedList_Count(LinkedList_t *instance)
+{
+   LinkedListNode_t *current = instance->head;
+   int count = 0;
+   while(current != NULL)
+   {
+      count++;
+      current = current->next;
+   }
+   return count;
+}
+
+void LinkedList_PushFront(LinkedList_t *instance, LinkedListNode_t *node)
+{
+   node = (LinkedListNode_t *)malloc(sizeof(LinkedListNode_t));
+   LinkedListNode_t *tempNode = instance->head;
+   instance->head = node;
+   node->next = tempNode;
+}
+
+void LinkedList_PushBack(LinkedList_t *instance, LinkedListNode_t *node)
+{
+   node = (LinkedListNode_t *)malloc(sizeof(LinkedListNode_t));
+   node->next = NULL;
+   if(instance->head == NULL)
+   {
+      instance->head = node;
+      return;
+   }
+   LinkedListNode_t *current = instance->head;
+   while(current->next != NULL) {
+      current = current->next;
+   }
+   current->next = node;
```

Comment:
Right now the body of this function is a bit difficult to read. I would recommend using whitespace or other means to help break up the separate thoughts in here.

For example, you could use whitespace to separate the `malloc`, the base case (empty list), and the normal case:

```c
   node = (LinkedListNode_t *)malloc(sizeof(LinkedListNode_t));
   node->next = NULL;
   
   if(instance->head == NULL)
   {
      instance->head = node;
      return;
   }
   
   LinkedListNode_t *current = instance->head;
   while(current->next != NULL) {
      current = current->next;
   }
   current->next = node;
```

Doing this is like using paragraphs in prose. It helps to separate "complete" thoughts and allow the reader to focus on one thought at a time.

You could also restructure a bit to make it more obvious that the base case and normal case are mutually exclusive:

```c
   node = (LinkedListNode_t *)malloc(sizeof(LinkedListNode_t));
   node->next = NULL;
   
   if(instance->head == NULL)
   {
      instance->head = node;
   }
   else
   {
      LinkedListNode_t *current = instance->head;
      while(current->next != NULL) {
         current = current->next;
      }
      current->next = node;
   }
```

Here the `if` and `else` blocks also serve to separate the complete thoughts of the base case and normal case, but also use the structure of the code to make it clear that they are not just separate concerns but mutually exclusive.